src/as-translation.c

```diff
@@ -221,6 +261,9 @@ as_translation_to_xml_node (AsTranslation *tr, AsContext *ctx, xmlNode *root)
 	n = xmlNewTextChild (root, NULL, (xmlChar*) "translation", (xmlChar*) priv->id);
 	xmlNewProp (n, (xmlChar*) "type",
 			(xmlChar*) as_translation_kind_to_string (priv->kind));
+
+	if (priv->source_locale != NULL)
+		xmlNewProp (n, (xmlChar*) "source_locale", (xmlChar*) priv->source_locale);
```

Comment:
Please use `as_xml_add_text_prop` here, it's a bit nicer for newly written code. So these whole two lines (you can drop the null check) would just become a single line:
```c
as_xml_add_text_prop (n, "source_locale", priv->source_locale);
```

Reply:
Nice